central/cve/image/v2/datastore/datastore_impl.go

```diff
@@ -0,0 +1,85 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/cve/common"
+	"github.com/stackrox/rox/central/cve/image/v2/datastore/search"
+	"github.com/stackrox/rox/central/cve/image/v2/datastore/store"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protocompat"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+type datastoreImpl struct {
+	storage  store.Store
+	searcher search.Searcher
+
+	cveSuppressionLock  sync.RWMutex
+	cveSuppressionCache common.CVESuppressionCache
+}
+
+func (ds *datastoreImpl) Search(ctx context.Context, q *v1.Query) ([]pkgSearch.Result, error) {
+	return ds.searcher.Search(ctx, q)
+}
+
+func (ds *datastoreImpl) SearchImageCVEs(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
+	return ds.searcher.SearchImageCVEs(ctx, q)
+}
+
+func (ds *datastoreImpl) SearchRawImageCVEs(ctx context.Context, q *v1.Query) ([]*storage.ImageCVEV2, error) {
+	cves, err := ds.searcher.SearchRawImageCVEs(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	return cves, nil
+}
+
+func (ds *datastoreImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
+	if q == nil {
```

Comment:
Why is this not needed for other search functions that take a query? Or is this check really needed at all?

Reply:
We don't.  Underlying calls will eventually do this.  will remove it.